src/exeptions/api.error.js

```diff
@@ -0,0 +1,32 @@
+export class ApiError extends Error {
+  constructor({ message, status, errors = {} }) {
+    super(message);
+
+    this.status = status;
+    this.errors = errors;
+  }
+
+  static badRequest(message, errors) {
+    return new ApiError({
+      message,
+      errors,
+      status: 400,
+    });
+  }
+
+  static unauthorize(errors) {
+    return new ApiError({
+      message: 'unauthorize user',
```

Comment:
The error message 'unauthorize user' should be 'unauthorized user' for grammatical correctness.